cmd/tellor/main.go

```diff
@@ -233,135 +250,145 @@ func newDisputeCmd(cmd *cli.Cmd) {
 	}
 }
 
-func mineCmd(cmd *cli.Cmd) {
-	remoteDS := cmd.BoolOpt("remote r", false, "connect to remote dataserver")
-	cmd.Action = func() {
-		// Create os kill sig listener.
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, os.Interrupt)
-		exitChannels := make([]*chan os.Signal, 0)
-
-		cfg := config.GetConfig()
-		var ds *ops.DataServerOps
-		if !cfg.EnablePoolWorker {
-			ErrorHandler(AddDBToCtx(*remoteDS), "initializing database")
-			if !*remoteDS {
-				ch := make(chan os.Signal)
-				exitChannels = append(exitChannels, &ch)
-
-				var err error
-				ds, err = ops.CreateDataServerOps(ctx, ch)
-				if err != nil {
-					log.Fatal(err)
-				}
-				// Start and wait for it to be ready.
-				if err := ds.Start(ctx); err != nil {
-					log.Fatal(err)
+func mineCmd(logger log.Logger) func(*cli.Cmd) {
+	return func(cmd *cli.Cmd) {
+		remoteDS := cmd.BoolOpt("remote r", false, "connect to remote dataserver")
+		cmd.Action = func() {
+			// Create os kill sig listener.
+			c := make(chan os.Signal, 1)
+			signal.Notify(c, os.Interrupt)
+			exitChannels := make([]*chan os.Signal, 0)
+
+			cfg := config.GetConfig()
+			var ds *ops.DataServerOps
+			if !cfg.EnablePoolWorker {
+				ErrorHandler(AddDBToCtx(*remoteDS), "initializing database")
+				if !*remoteDS {
+					ch := make(chan os.Signal)
+					exitChannels = append(exitChannels, &ch)
+
+					var err error
+					ds, err = ops.CreateDataServerOps(ctx, logger, ch)
+					if err != nil {
+						level.Error(logger).Log("msg", "error creating data server", "err", err)
+						os.Exit(1)
+					}
+					// Start and wait for it to be ready.
+					if err := ds.Start(ctx, logger); err != nil {
+						level.Error(logger).Log("msg", "error starting data server", "err", err)
```

Comment:
I failed start should also exit here no?

```suggestion
						level.Error(logger).Log("msg", "error starting data server", "err", err)
						os.Exit(1)
```